cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -215,14 +216,60 @@ func generateMountsModeRun() error {
 		base := filepath.Join(dataDir, "system-data", dirs.SnapBlobDir, modeEnv.Base)
 		fmt.Fprintf(stdout, "%s %s\n", base, filepath.Join(runMnt, "base"))
 	}
+
+	// 2.3 mount kernel
 	isKernelMounted, err := osutilIsMounted(filepath.Join(runMnt, "kernel"))
 	if err != nil {
 		return err
 	}
 	if !isKernelMounted {
-		// XXX: do we need to cross-check the booted/running kernel vs the snap?
```

Comment:
Could we extract this part of the code (with backward compatibility) into it's own PR? Presumably it will be smaller and easier to review and we could get it into initramfs quicker. Then we could have spread test. Or is that undesired for some reason?

Reply:
As discussed over IRC, we will leave this bit here, and in a followup I will split out the logic into a helper